### PR TITLE
[DO NOT MERGE] Split gtest and rostest cpp coverage reports.

### PR DESCRIPTION
**Pull request by knorth55**
merge after #17 
this PR split gtest and rostest cpp coverage reports.

see: https://github.com/GITAI/gitai_tools/pull/290#issue-1862880840

### CODE REVIEW

**Review by garaemon — CHANGES_REQUESTED** [review]
This PR looks too large. Could you separate into some smaller pull requests? Or could you organize commit history? For example, "format" commit includes some changes not only about format but also implementation.

**Comment by knorth55** [comment]
Format commit is actually just formatting with linter.
but the modification was too large so that git cannot track the change and shows as huge difference
#17

**Comment by garaemon** [comment]
Oh, I see.

**Comment by knorth55** [comment]
Without this formatting, this PR will be easier than it was.
but I change the dependency structure drastically to aplit gtest and rostest.
I will try to make this pr simple.

File: cmake/Modules/CodeCoverage.cmake
Comment (garaemon) [inline cmake/Modules/CodeCoverage.cmake:418]: Could you separate this `add_custom_command` into a few commands? I think using `if` in multiple `COMMAND` is too complicated.

File: cmake/Modules/CodeCoverage.cmake
Comment (garaemon) [inline cmake/Modules/CodeCoverage.cmake:242]: ditto

File: cmake/Modules/CodeCoverage.cmake
Comment (garaemon) [inline cmake/Modules/CodeCoverage.cmake:459]: ditto

File: cmake/Modules/CodeCoverage.cmake
Comment (garaemon) [inline cmake/Modules/CodeCoverage.cmake:520]: Please add a comment about why you need to run `_create_coverage_report_${PROJECT_NAME}_gtest_cpp` first.

**Review by garaemon — CHANGES_REQUESTED** [review]
I didn't leave all the comments but please try to separate big `add_custom_command`s into a smaller commands. If it looks messy, let's rethink it.

From next time, please separate PRs for refactoring, large formatting and function addition.

**Comment by knorth55** [comment]
@garaemon I'm sorry. I thought I requested a review for #17 first, but I didn't.
I will wait for #17 review before merging this PR.

File: cmake/Modules/CodeCoverage.cmake
Comment (knorth55) [reply cmake/Modules/CodeCoverage.cmake:418]: the command is too complicated to me, too.
But I also don't want to split the command and make a lot of custom target.
Now I'm thinking of rewriting the commands into python scripts, and keep the target dependency simple.
@garaemon what do you think?

My plan is.
- move this script to gitai_code_coverage
- covert the add_custom_commands to python script

**Comment by knorth55** [comment]
Now I rebased `origin/master` to make the PR's diff simpler than it was.

**Comment by knorth55** [comment]
@garaemon 

Thank you for your review.

I split this PR into 5 PRs.
- #19 
- #20 
- #21 
- #22 
- #23 

It depends on each other, so can you review in this order?
#19, #20 -> #21 -> #22 -> #23

File: cmake/Modules/CodeCoverage.cmake
Comment (knorth55) [reply cmake/Modules/CodeCoverage.cmake:418]: @garaemon 

I splitted this PR into #19 #20 #21 #22 #23.
After #23 is merged, I will work changing this scripts into python.

**Pull request closed without merging**
